modules/pricer/src/main/java/com/opengamma/strata/pricer/calibration/CurveBuildingBlock.java

```diff
@@ -0,0 +1,412 @@
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.google.common.collect.ImmutableList;
+import com.opengamma.strata.market.curve.CurveName;
+
+/**
+ * Information about a block of curves that was built together.
+ * <p>
+ * This provides details of the names of the curves and the number of parameters they have.
+ * This is used in curve calibration to create and split a single array of all parameters from all curves.
+ */
+@BeanDefinition(builderScope = "private")
+public final class CurveBuildingBlock
+    implements ImmutableBean, Serializable {
+
+  /**
+   * The empty block.
+   */
+  private static final CurveBuildingBlock EMPTY = new CurveBuildingBlock(ImmutableList.of());
+
+  /**
+   * The curve parameter sizes.
+   * <p>
+   * The entries are used to define the curve order and the number of parameters.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final ImmutableList<CurveParameterSize> data;
+
+  //-------------------------------------------------------------------------
+  /**
+   * Obtains an empty block.
+   * 
+   * @return the empty block
+   */
+  public static CurveBuildingBlock empty() {
+    return EMPTY;
+  }
+
+  /**
+   * Obtains a block from a map of curve name to curve parameter size.
+   * 
+   * @param data  the block data
+   * @return the block
+   */
+  public static CurveBuildingBlock of(List<CurveParameterSize> data) {
+    return new CurveBuildingBlock(data);
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Gets the total number of curves.
+   * 
+   * @return the number of curves
+   */
+  public int getCurveCount() {
+    return data.size();
+  }
+
+  /**
+   * Gets the total number of parameters.
+   * 
+   * @return the number of curves
+   */
+  public int getTotalParameterCount() {
+    int total = 0;
+    for (CurveParameterSize entry : data) {
+      total += entry.getParameterCount();
+    }
+    return total;
+  }
+
+  /**
+   * Finds the start index of the specified curve.
+   * <p>
+   * The result represents the start in index in the combined array of all parameters.
+   * 
+   * @param name  the curve name
+   * @return the start index
+   * @throws IllegalArgumentException if the curve cannot be found
+   */
+  public int getStart(CurveName name) {
+    int start = 0;
+    for (CurveParameterSize entry : data) {
+      if (entry.getName().equals(name)) {
+        return start;
+      }
+      start += entry.getParameterCount();
+    }
+    throw new IllegalArgumentException("Unable to find data for curve: " + name);
+  }
+
+  /**
+   * Finds the number of parameters of the specified curve.
+   * 
+   * @param name  the curve name
+   * @return the number of parameters
+   * @throws IllegalArgumentException if the curve cannot be found
+   */
+  public int getParameterCount(CurveName name) {
+    for (CurveParameterSize entry : data) {
```

Comment:
return data.stream()
            .filter(entry -> entry.getName().equals(name))
            .findFirst()
            .map(CurveParameterSize::getParameterCount)
            .orElseThrow(() -> new IllegalArgumentException("Unable to find data for curve: " + name));